src/test/spec/unified_runner/operation.rs

```diff
@@ -2640,6 +2643,62 @@ impl TestOperation for CreateEntities {
     }
 }
 
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase", deny_unknown_fields)]
+pub(super) struct Download {
+    id: Bson,
+}
+
+impl TestOperation for Download {
+    fn execute_entity_operation<'a>(
+        &'a self,
+        id: &'a str,
+        test_runner: &'a TestRunner,
+    ) -> BoxFuture<'a, Result<Option<Entity>>> {
+        async move {
+            let bucket = test_runner.get_bucket(id).await;
+            let mut buf: Vec<u8> = vec![];
+            bucket
+                .download_to_futures_0_3_writer(self.id.clone(), &mut buf)
```

Comment:
I know we talked at some point about figuring out a way to get coverage of the various upload/download methods via the unified tests. but maybe using the futures methods makes sense here since (I think) they end up leading us to call through to the other versions of the method anyway?
if that's the case I think it might be worth explaining in a comment here why we chose this variant of `download` (likewise for `DownloadByName`)

Reply:
Yeah, I chose to use the `futures` API because it runs through the `tokio` API, so we're getting coverage for both here. It seems like we're leaning towards only having one API but I'll add a comment here explaining why `futures` is used if we decide to keep both.